pylossless/pipeline.py

```diff
@@ -798,68 +806,185 @@ def find_breaks(self):
+        is flagged. A cutoff threshold other than 20% can be provided, if set
+        in the config.
+
+        WARNING: the default threshold of 50 microvolts may not be appropriate
+        for a particular dataset or data file, as the baseline voltage variance
+        is affected by the impedance of the system that the data was recording
+        on. You may need to assess a more appropriate value for your own data.
+        """
+        epochs = self.get_epochs()
+        above_threshold = _threshold_volt_std(epochs,
+                                              flag_dim=flag_dim,
+                                              threshold=threshold)
+        dim = {'ch': "channels", 'epoch': "epochs"}
+        print(f'🧑‍🔧 flag_{dim[flag_dim]}_fixed_threshold report!!!',
+              above_threshold)
+        self.flags[flag_dim].add_flag_cat('volt_std', above_threshold, epochs)
+
+    def find_outlier_chs(self):
+        """Detect outlier Channels to leave out of rereference."""
         # TODO: Re-use _detect_outliers here.
-        # Window the continuous data
-        # logging_log('INFO', 'Windowing the continuous data...');
-        epochs_xr = epochs_to_xr(self.get_epochs(), kind="ch")
+        logger.info("🔍 Detecting channels to leave out of reference..")
+        epochs_xr = epochs_to_xr(self.get_epochs(rereference=False), kind="ch")
 
         # Determines comically bad channels,
         # and leaves them out of average rereference
-        trim_ch_sd = variability_across_epochs(epochs_xr)
-        # std across epochs for each chan; shape (chans, time)
-
+        trim_ch_sd = epochs_xr.std('time')
         # Measure how diff the std of 1 channel is with respect
         # to other channels (nonparametric z-score)
         ch_dist = trim_ch_sd - trim_ch_sd.median(dim="ch")
         perc_30 = trim_ch_sd.quantile(0.3, dim="ch")
         perc_70 = trim_ch_sd.quantile(0.7, dim="ch")
-        ch_dist /= perc_70 - perc_30  # shape (chans, time)
+        ch_dist /= perc_70 - perc_30  # shape (chans, epoch)
 
-        mean_ch_dist = ch_dist.mean(dim="time")  # shape (chans)
+        mean_ch_dist = ch_dist.mean(dim="epoch")  # shape (chans)
 
         # find the median and 30 and 70 percentiles
         # of the mean of the channel distributions
         mdn = np.median(mean_ch_dist)
         deviation = np.diff(np.quantile(mean_ch_dist, [0.3, 0.7]))
 
-        bad_ch_names = mean_ch_dist.ch[mean_ch_dist > mdn+6*deviation]
-        self.flagged_chs.add_flag_cat(kind='outliers',
-                                      bad_ch_names=bad_ch_names)
+        return mean_ch_dist.ch[mean_ch_dist > mdn+6*deviation].values.tolist()
 
-        # TODO: Verify: It is unclear this is necessary.
-        # get_epochs() is systematically rereferencing and
-        # all steps (?) uses the get_epochs() function
-        self.flagged_chs.rereference(self.raw)
+    def flag_channels_fixed_threshold(self, threshold=5e-5):
```

Comment:
This is a function we coded together (I asked for your help) to flag channels based on a fixed threshold of the voltage-var stdev (per the lossless paper.)